spell/spell-reviewer-mainnet-checklist.md

```diff
@@ -1,369 +1,372 @@
+* IF debt ceiling changes are present in the instructions
+  * IF adjusted collateral type (`ilk`) have [AutoLine](https://github.com/makerdao/dss-auto-line/tree/master) enabled (`MCD_IAM_AUTO_LINE`)
+    * IF collateral debt ceiling requested to be `0`
+      * [ ] Collateral is removed from AutoLine (`MCD_IAM_AUTO_LINE`) via [`DssExecLib.removeIlkFromAutoLine(ilk)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L668)
+      * [ ] The instruction to remove from AutoLine (`MCD_IAM_AUTO_LINE`) is present in the Exec Sheet
+      * [ ] Collateral debt ceiling is set to `0` via [`DssExecLib.setIlkDebtCeiling(ilk, amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L611)
+      * [ ] Global debt ceiling (`vat.Line`) is updated accordingly, UNLESS specifically instructed not to
+    * IF `AutoLine` parameters are updated
+      * [ ] Either is used, depending on the instruction:
+        * [`DssExecLib.setIlkAutoLineDebtCeiling(ilk, amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L658)
+        * [`DssExecLib.setIlkAutoLineParameters(ilk, amount, gap, ttl)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L648)
+  * IF collateral debt ceiling (`vat.ilk.line`) is updated
+    * [ ] Collateral type (`ilk`) have [`AutoLine`](https://github.com/makerdao/dss-auto-line/tree/master) disabled previously or in the spell
+    * [ ] Either is used, depending on the instruction:
+        * [`DssExecLib.increaseIlkDebtCeiling(ilk, amount, global)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L621C14-L621C36)
+        * [`DssExecLib.decreaseIlkDebtCeiling(ilk, amount, global)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L634)
+        * [`DssExecLib.setIlkDebtCeiling(ilk, amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L611)
+    * [ ] Global debt ceiling (`vat.Line`) is updated accordingly, UNLESS specifically instructed not to, via either:
+        * `global` set to `true` in `increaseIlkDebtCeiling`/`decreaseIlkDebtCeiling`
+        * [`DssExecLib.setGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L428)
+        * [`DssExecLib.increaseGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L436)
+        * [`DssExecLib.decreaseGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L445C14-L445C39)
+* IF onboarding is present
+  * [ ] Insert and follow relevant checklists here
+    * [Collateral Onboarding](./collateral-onboarding-checklist.md)
+    * [RWA Onboarding](./rwa-onboarding-checklist.md)
+    * [Teleport Onboarding](./teleport-onboarding-checklist.md)
+* IF collateral offboarding is present in the spell
+  * 1st stage collateral offboarding
```

Comment:
I avoided using `for each`, because then 99% of the checks would have it. I propose to leave it out or at least specify it as a meta-rule somewhere else.